devlib/target.py

```diff
@@ -2995,7 +2992,7 @@ def __init__(self,
         adb_conn_params = ['device', 'adb_server', 'timeout']
         self.android_connection_settings = {}
         for setting in adb_conn_params:
-            if connection_settings.get(setting, None):
+            if setting in connection_settings:
```

Comment:
this is not equivalent. If `setting` is available in `connection_settings` with a falsy value, the `setting in connection_settings` will still enter the branch.

OTH if we decide that we want falsy values to be propagated, we can do it with a comprehension:
```
self.android_connection_settings.update(
  (key, value)
  for key, value in connection_settings.items()
  if key in adb_conn_params
)
```

Reply:
Ok sounds good thanks